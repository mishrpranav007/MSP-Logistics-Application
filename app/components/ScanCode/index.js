import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Linking,
  Image,
  TouchableOpacity
} from 'react-native';
import QRCodeScanner from 'react-native-qrcode-scanner';
import { RNCamera } from 'react-native-camera';
import flashImage from 'app/assets/images/flash.png';
import cancelImage from 'app/assets/images/cancel.png';
import PropTypes from 'prop-types';
import Style from '../../style/index';

export default class ScanCode extends Component {
  constructor(props) {
    super(props);
    this.state = {
      torchEnable: false
    };
  }

  onSuccess = e => {
    Linking.openURL(e.data);
  };

  onTorchToggle = () => {
    const { torchEnable } = this.state;
    this.setState({ torchEnable: !torchEnable });
  };

  onCancel = () => {
    this.props.navigation.goBack();
  };

  render() {
    const { torchEnable } = this.state;
    return (
      <View style={styles.container} testID="scan-code">
        <View style={styles.headerIcons}>
          <TouchableOpacity onPress={this.onTorchToggle}>
            <Image source={flashImage} style={styles.imageContainer}></Image>
          </TouchableOpacity>
          <TouchableOpacity onPress={this.onCancel}>
            <Image
              source={cancelImage}
              style={styles.cancelImageContainer}
            ></Image>
          </TouchableOpacity>
        </View>
        <QRCodeScanner
          testID="scan-code-qrcode-scanner"
          containerStyle={styles.cameraContainer}
          onRead={this.onSuccess}
          showMarker
          reactivate
          checkAndroid6Permissions
          flashMode={
            torchEnable
              ? RNCamera.Constants.FlashMode.torch
              : RNCamera.Constants.FlashMode.off
          }
          customMarker={
            <View style={styles.rectangleContainer}>
              <View style={styles.rectangle} />
            </View>
          }
          cameraStyle={styles.cameraView}
          cameraProps={{ ratio: '1:1' }}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Style.BLACK_COLOR,
    opacity: 0.99
  },
  cameraContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Style.BLACK_COLOR,
    marginTop: Style.em(11)
  },
  cameraView: {
    justifyContent: 'center',
    alignItems: 'center',
    overflow: 'visible',
    height: Style.em(14),
    width: Style.em(13),
    borderRadius: 10,
    marginTop: Style.em(-0.7)
  },
  rectangleContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
    marginTop: Style.em(-0.4)
  },
  rectangle: {
    alignItems: 'center',
    overflow: 'visible',
    justifyContent: 'center',
    height: Style.em(17.5),
    width: Style.em(17.5),
    borderWidth: 2,
    borderColor: Style.GREENISH_TEAL,
    borderRadius: 40,
    backgroundColor: 'transparent'
  },

  imageContainer: {
    marginTop: Style.em(1),
    marginLeft: Style.em(1)
  },
  headerIcons: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  cancelImageContainer: {
    marginRight: Style.em(1),
    marginTop: Style.em(1.2)
  }
});
ScanCode.propTypes = {
  navigation: PropTypes.object
};
