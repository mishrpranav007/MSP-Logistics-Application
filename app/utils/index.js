import { Platform, Dimensions } from 'react-native';

export const isAndroid = () => Platform.OS !== 'ios';
export const getScreenWidth = () => Dimensions.get('window').width;
export const getScreenHeight = () => Dimensions.get('window').height;

export const getPlatformString = () => (isAndroid() ? 'android' : 'ios');
export const getFontFamily = font =>
  isAndroid() ? `Futura-${font}` : `FuturaStd-${font}`;

export const isScreenSmall = () => getScreenHeight() <= 550;
