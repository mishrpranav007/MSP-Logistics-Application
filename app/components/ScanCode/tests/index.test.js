import React from 'react';
import { renderWithIntl } from 'app/utils/testUtils';
import QRCodeScanner from 'react-native-qrcode-scanner';
import ScanCode from '../index';

describe('<ScanCode />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<ScanCode />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 scancode screen component', () => {
    const { getAllByTestId } = renderWithIntl(<ScanCode />);
    expect(getAllByTestId('scan-code').length).toBe(1);
  });
  it('should contain 1 qrcodescanner contained  within', () => {
    const { getAllByTestId } = renderWithIntl(
      <ScanCode>
        <QRCodeScanner />
      </ScanCode>
    );
    expect(getAllByTestId('scan-code').length).toBe(1);
  });
});
