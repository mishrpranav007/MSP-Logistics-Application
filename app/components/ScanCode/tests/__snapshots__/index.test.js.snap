// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScanCode /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(0,0,0)",
          "flex": 1,
          "opacity": 0.99,
        }
      }
      testID="scan-code"
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/images/flash.png",
              }
            }
            style={
              Object {
                "marginLeft": 25.142857142857142,
                "marginTop": 25.142857142857142,
              }
            }
          />
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/images/cancel.png",
              }
            }
            style={
              Object {
                "marginRight": 25.142857142857142,
                "marginTop": 30.17142857142857,
              }
            }
          />
        </TouchableOpacity>
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(0,0,0)",
              "flex": 1,
              "justifyContent": "center",
              "marginTop": 276.57142857142856,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 2,
                "justifyContent": "center",
                "width": 750,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            ...
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 2,
                "justifyContent": "center",
                "width": 750,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
